### DISCUSSION TITLE
Add version constraints to vcpkg ports

### DISCUSSION THREAD

**Discussion by dg0yt** [microsoft/vcpkg#21872] [Ideas]
I recently learned that vcpkg port can use `>=` version constraints. IMO this should become mandatory for new features in maintainer ports. We see a number of issues due to outdated installed maintainer ports. E.g.
- Option `MAYBE_UNUSED_VARIABLES` from port `vcpkg-cmake`, #21862.
- Boost ports need at least the same version of port `boost-modular-build-helper`.

Given that maintainer ports are host dependencies, a naive `vcpkg remove ...` without host triplet doesn't help.

**Comment by thxkiwi** [comment]
Isn't this handled in manifests via `overrides`?

https://vcpkg.readthedocs.io/en/latest/users/versioning/

**Reply by dg0yt** [reply]
No. It is entirely unrelated to user-side versioning.
